### PR TITLE
Treat =< and => as == in scripts

### PR DESCRIPTION
**Pull request by Allofich**
For https://bugs.openmw.org/issues/3744

### CODE REVIEW

**Comment by Allofich** [comment]
I just found that `=<<` also works as "equals," and so does `=<<>,` but `=<<>6` fails, so it looks like you you can have an arbitrary number of < or > in there as long as it doesn't have anything else. But I don't think we need to worry about these cases.

**Comment by Allofich** [comment]
Also see my comment in the bug. Should this be fixed?

**Comment by Capostrophic** [comment]
IMO OpenMW basically shouldn't crap pants when it stumbles upon those typos. Ignoring them (say, to copy Morrowind engine behavior of "i dunno so let it be "equal"" like your solution) instead of outright dropping script execution doesn't really encourage bad plugin design. However, yes, Kortunov is right, there definitely should be a warning - at least for OpenMW-CS' script verifier.

**Comment by Allofich** [comment]
Updated it with a warning message

File: components/compiler/scanner.cpp [hunk 1]
```diff
@@ -408,6 +408,11 @@ namespace Compiler
                     special = S_cmpEQ;
                 else if (c=='=')
                     special = S_cmpEQ;
+                else if (c == '>' || c == '<')  // Treat => and =< as ==
+                {
+                    special = S_cmpEQ;
+                    mErrorHandler.warning ("reversed <= or >= operator, treating it as ==", mLoc);
```
Comment (scrawl) [inline components/compiler/scanner.cpp]: To be more specific, I'd use:
"invalid operator =" + c + ", treating it as =="

Comment (Allofich) [inline components/compiler/scanner.cpp]: OK I used your suggestion.

**Pull request merged**